app/src/main/java/tech/jpco/qen/view/QenPage.kt

```diff
@@ -0,0 +1,154 @@
+        .filterForActionAndTime(50)
+        .map {
+            val (normX, normY) = normalize(it.x, it.y)
+            val type = when (it.action) {
+                MotionEvent.ACTION_DOWN -> TouchEventType.TouchDown
+                MotionEvent.ACTION_UP -> TouchEventType.TouchUp
+                MotionEvent.ACTION_MOVE -> TouchEventType.TouchMove
+                else -> throw IllegalArgumentException(
+                    "Tried to turn a MotionEvent other than Down, Up, or " +
+                            "Move into a DrawPoint!"
+                )
+            }
+            DrawPoint(normX, normY, type)
+        }
+        .filterForMinDistance { x, y -> Math.abs(x) + Math.abs(y) < normDistance(10) }
+
+    val arStream: Observable<Float> = layoutChanges().map { ar }
+
+    fun drawSegment(new: DrawPoint, shouldInvalidate: Boolean = true) {
+        val (denormX, denormY) = denormalize(new.x, new.y)
+        val denormNew = new.copy(x = denormX, y = denormY)
+        if (lastPoint != null && denormNew.type != TouchEventType.TouchDown) {
+            lastPoint?.apply {
+                bufferCanvas.drawLine(x, y, denormNew.x, denormNew.y, paint)
+                if (shouldInvalidate) invalidate()
+            }
+        }
+        lastPoint = if (denormNew.type == TouchEventType.TouchUp) null else denormNew
+    }
+
+    fun clearPage() {
+        Log.d(TAG, "clearPage called")
+        bufferBitmap.eraseColor(Color.TRANSPARENT)
+        invalidate()
+    }
+
+    fun drawPage(list: List<DrawPoint>, newAR: Float) {
+        clearPage()
+        list.forEach { drawSegment(it, false) }
+        Log.d(TAG, "drawPage completing")
+        invalidate()
+    }
+
+    //comment-detritus from aborted aspect-ratio implementation
+    private fun normalize(x: Float, y: Float): Pair<Float, Float> =
+        /*if (ar < 1)*/ Pair(x / mWidth, y / mHeight)
+    /*else Pair(y / mHeight, x / mWidth)*/
+
+    private fun normDistance(pixels: Int) = pixels / (Math.sqrt((mHeight * mHeight + mWidth * mWidth).toDouble()))
+
+    //comment-detritus from aborted aspect-ratio implementation
+    private fun denormalize(relX: Float, relY: Float) =
+        /*if (ar < 1)*/ Pair(relX * mWidth, relY * mHeight)
+    /*else Pair(relY * mHeight, relX * mWidth)*/
+
+    private fun Observable<MotionEvent>.filterForActionAndTime(minMilli: Long): Observable<MotionEvent> =
+        publish { multiStream ->
+            Observable.merge(
+                multiStream
+                    .filter {
+                        it.action == MotionEvent.ACTION_DOWN ||
+                                it.action == MotionEvent.ACTION_UP
+                    },
+                multiStream
+                    .filter {
+                        it.action == MotionEvent.ACTION_MOVE
+                    }
+                    .throttleFirst(minMilli, TimeUnit.MILLISECONDS)
+
+            )
+        }
+
+
+    //scan() emits Pair(the last valid point, the new point IFF it's valid)
```

Comment:
Noted in general, but in this case it's a comment on a private function, because I was looking at the function and scratching my head, and my internal policy for comments is that at a minimum I should document any code that I had to figure out a second time!